pkg/batcher/controllers/default.go

```diff
@@ -0,0 +1,307 @@
+	batcherHandler *BatcherHandler = nil
+	mutex sync.Mutex
+)
+
+type MainController struct {
+	beego.Controller
+}
+
+type Request struct {
+	Instances []interface{} `json:"instances"`
+}
+
+type Input struct {
+	ContextInput *context.Context `json:"contextInput"`
+	Instances *[]interface{} `json:"instances"`
+	ChannelOut *chan Response `json:"channelOut"`
+}
+
+type InputInfo struct {
+	ChannelOut *chan Response `json:"channelOut"`
+	Index []int `json:"index"`
+}
+
+type Response struct {
+	BatchID string `json:"batchId"`
+	Predictions []interface{} `json:"predictions"`
+}
+
+type ResponseError struct {
+	Message string `json:"message"`
+}
+
+type Predictions struct {
+	Predictions []interface{} `json:"predictions"`
+}
+
+type BatcherInfo struct {
+	MaxBatchsize int `json:"maxBatchsize"`
+	MaxLatency float64 `json:"maxLatency"`
+	BatchID string `json:"batchId"`
+	Instances []interface{} `json:"instances"`
+	Predictions Predictions `json:"predictions"`
+	Info map[*context.Context] InputInfo `json:"info"`
+	Start time.Time `json:"start"`
+	Now time.Time `json:"now"`
+	CurrentInputLen int `json:"currentInputLen"`
+}
+
+type BatcherHandler struct {
+	Log              logr.Logger
+	Port             string
+	SvcHost          string
+	SvcPort          string
+	MaxBatchsize     int
+	MaxLatency       float64
+	Path             string
+	ContentTpye      string
+}
+
+func New(log logr.Logger, port string, svcHost string, svcPort string,
+	maxBatchsize int, maxLatency float64) {
+	batcherHandler = &BatcherHandler{
+		Log:              log,
+		Port:             port,
+		SvcHost:          svcHost,
+		SvcPort:          svcPort,
+		MaxBatchsize:     maxBatchsize,
+		MaxLatency:       maxLatency,
+		Path:             "",
+		ContentTpye:      "",
+	}
+}
+
+func GetNowTime() time.Time {
+	return time.Now().UTC()
+}
+
+func GenerateUUID() string {
+	if uuidStr, err := uuid.NewV4(); err == nil {
+		return uuidStr.String()
+	}
+	return ""
+}
+
+func CallService() {
+	var errStr string
+	url := fmt.Sprintf("http://%s:%s%s", batcherHandler.SvcHost, batcherHandler.SvcPort, batcherHandler.Path)
+	jsonStr, _ := json.Marshal(Request{
+		batcherInfo.Instances,
+	})
+	batcherHandler.Log.Info("CallService", "URL", url)
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		errStr = fmt.Sprintf("NewRequest create fail: %v", err)
+		panic(errStr)
+	}
+	req.Header.Add("Content-Type", batcherHandler.ContentTpye)
+	defer req.Body.Close()
+	client := &http.Client{Timeout: 5 * time.Second}
+	resp, err := client.Do(req)
+	if err != nil {
+		errStr = fmt.Sprintf("NewRequest send fail: %v", err)
+		panic(errStr)
+	}
+	defer resp.Body.Close()
+	result, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		errStr = fmt.Sprintf("Response read fail: %v", err)
+		panic(errStr)
+	}
+	err = json.Unmarshal(result, &batcherInfo.Predictions)
+	if err != nil {
+		errStr = fmt.Sprintf("Response unmarshal fail: %v, %s", err, string(result))
+		panic(errStr)
+	}
```

Comment:
why do we need to unmarshal and then marshal again?

Reply:
Uh, that is redundant. I will remove it.